core/controllers/learner_dashboard.py

```diff
@@ -146,20 +146,20 @@ class LearnerDashboardFeedbackThreadHandler(base.BaseHandler):
     """Gets all the messages in a thread."""
 
     @acl_decorators.can_access_learner_dashboard
-    def get(self, exploration_id, thread_id):
+    def get(self, thread_id):
         """Handles GET requests."""
-        messages = feedback_services.get_messages(
-            exploration_id, thread_id)
+        messages = feedback_services.get_messages(thread_id)
         author_ids = [m.author_id for m in messages]
         authors_settings = user_services.get_users_settings(author_ids)
 
         message_ids = [m.message_id for m in messages]
         feedback_services.update_messages_read_by_the_user(
-            self.user_id, exploration_id, thread_id, message_ids)
+            self.user_id, thread_id, message_ids)
 
         message_summary_list = []
-        suggestion = feedback_services.get_suggestion(exploration_id, thread_id)
+        suggestion = feedback_services.get_suggestion(thread_id)
 
+        exploration_id = thread_id.split('.')[0]
```

Comment:
This needs a TODO to generalize later, right? We should probably make this a method on the domain object or something (and I'm not sure if we need subclasses for Thread domain objects as well, each with their own "extract foo from ID" method).

Reply:
I'll make this a method on the domain object. With the plan of generalizing, I don't think we need subclasses as all IDs will have the same pattern and we can extract entity ID from thread_id directly. The corresponding domain method will need to be changed to get the correct parameter from the split string. 

The generalization step would involve changing exploration_id to entity_id. And the addition of an extra value in the ID -- the entity_type. But since the structure and behavior for threads shouldn't depend on the entity type, I don't think we need subclassing here.

Reply:
Ah, I wrote "todo" originally, but I would actually suggest generalizing now (after making the rest of the comments below). A method on the domain object sounds good as long as it is clearly generalizable to stuff beyond "exploration". And I agree that subclassing might not be needed.

Reply:
Ok. I'll make the changes.

Reply:
Done.